### ISSUE TITLE
Wrong filesize downloaded with S3 enabled

### ISSUE THREAD

**Issue by dvrcthewrld** [graphpaperpress/Sell-Media#801]
With current versions, when the S3extension is enabled, the file that the buyer downloads (having purchased the original size) is actually the large size as defined in Settings>Media.

To fix this… In inc/class-downloads.php, around line 42

```
 if ( $verified ) {                
                // If S3 extension, then get attachment url
                if ( class_exists( 'SellMediaS3' ) ) {
                    $file = wp_get_attachment_url( $attachment_id );

	        nocache_headers();
                header( "Robots: none" );
                header( "Content-Description: File Transfer" );
                header( "Content-Disposition: attachment; filename=".array_shift(explode('?', basename($file))));
	        readfile($file);

        	do_action( 'sell_media_after_successful_download', $product_id );
                exit();
                } else {

                $file = Sell_Media()->products->get_protected_file( $product_id, $attachment_id );
```

**Comment by ralf221** [comment]
Hello dvrcthewrld, thanks for helping. we tried your code it works with the regular purchased download. So you buy a single picture and get a download link - this works. But if you are a subscriber and you start to download - you still end up at the low resolution files. Any Idea to help please?

**Comment by dvrcthewrld** [comment]
Sorry, @ralf221, I don't operate with subscriptions, nor do I want to.  I'm sure that the relevant code is within the subscriptions extension.

Also, I have stopped use this extension as well.  I now use a third-party WP extension to upload all media to S3.  It's all private and inaccessible, but it serves as the repository for my CloudFront distribution.  And my CF distribution requires signed URLs, with the exception of downsized, watermarked images.  And I have written custom code to handle serving up signed CloudFront URLs to buyers.

If the GPM folks are interested, I'm happy to talk with them about it.

**Comment by ralf221** [comment]
Hello thanks so much for your reply. My website is up and running and I did upload a whole lott of pictures with descriptions and so on. A lott of work. So I can't just swich to another service. Can I get you something if you find a solution for me? Thank a lott Ralf  This is the site: www.thexlargeimage.com

**Comment by WEBJONGENS** [comment]
We having the same issue. Wordpress version: 4.8.3 SellMedia version 2.4.1.
After enabling the S3 Amazon storage option, all images are default media size of 1024 px. Customers are not able to download the higher resolutions anymore. I tried editing the file inc/class-downloads.php, around line 42 as suggested, but our file looks some kind different than your file, so the customisation is not working. Please provide us with a solution.

**Comment by rickalday** [comment]
Fixed in dev branch e20d0e30f23e4725ac862bd718cb785a91a6ec08
